### ISSUE TITLE
Question writing plugins about the AlbumInfo

### ISSUE THREAD

**Issue by HOZHENWAI** [beetbox/beets#4896]
### Original post https://github.com/beetbox/beets/discussions/4892

<div type='discussions-op-text'>

<sup>Originally posted by **HOZHENWAI** September  3, 2023</sup>
Hello, I'm currently updating an old plugin that I made for beets to use another metadata website for videogame albums.
I have some issues with the album_id and the artist id field in the AlbumInfo class. 
Given the documentation, it looks like there are 5 required fields:
- album
- album_id (a music brainz id is required? even though i'm not using musicbrainz)
- artist
- artist_id (what is this?)
- tracks_id
What is the correct way to fill those information? I've seen the code for discogs and spotify and i'm surprised by the following design choices:
- there is a specific `discogs_albumid` and `discogs_artistid` from the AlbumInfo, (therefore isn't the base beets class assuming the existence of the plugin?) doesn't that mean that if you want to write a metadata plugin, you will have to write new pr to the beets project and create migration scripts? that does not sound scalable...
- the spotify plugin fill a special `spotify_album_id` that does not exist for the albuminfo, i'm not sure what are the consequences of that (will it serialize when adding into the database? what about when searching?)
- in the spotify plugin, the `album_id` is filled with the `spotify_album_id`, which i'm not sure is an uuid?
I'm asking these questions cause I want to write a clean plugin and the old solution I did, which was to fill the album_id with a custom string did not look to be the proper way</div>

**Comment by arsaboo** [comment]
These are great questions and I would also love some clarity on these. Here's my impression of how things work. 
`album_id` is required, but it doesn't have to be an MBID. For example, in the Spotify plugin, I have used `album_id=spotify_id`. Similarly, `artist_id` and `track_id` can be any unique identifier. For most of the plugins I wrote, I just used the same values as those obtained from the source for these fields. In addition, I included the same in the source-specific field (e.g., spotify_track_id). I do recall @JOJ0 using a built-in function to generate `artist_id`. 

For non-MB sources, I honestly, did not give these fields much thought, but it will be nice to have some clarity and recommended best practices.
